### PR TITLE
Implement FigureMover

### PR DESCRIPTION
**Pull request by alena-bartosh**

### CODE REVIEW

File: src/figure_mover.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        return new_points, new_rotation
+
+    @staticmethod
+    def move_right(figure: Figure) -> NewPoints:
+        new_points: NewPoints = []
+
+        for point in figure.points:
+            new_points.append(Point(point.x, point.y + 1))
+
+        return new_points
```
Comment (s373r) [inline src/figure_mover.py]: extract to ... 
```py
def __move(figure: Figure, direction: Point) -> NewPoints:
```

... and use it inside `move_right()`,  `move_left()`, `move_down()`

File: src/figure_mover.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+from point import Point, Points
+from rotation_kind import RotationKind
+
+NewPoints = Points
+NewRotation = RotationKind
+RotationResult = Tuple[NewPoints, NewRotation]
+
+
+class FigureMover:
+    rotation_figure_points_mapping: dict[RotationKind, dict[FigureKind, Points]] = {
```
Comment (s373r) [inline src/figure_mover.py]: please `dict[FigureKind, dict[RotationKind, Points]]`

**Review by Serg-Bartosh — DISMISSED** [review]
“Every person on Earth is a solution to someone’s problem,” my wise grandmother once said.
I was very surprised by her words.
“You are the solution to someone's problem,” she repeated.
Stated:
- The gift that was given to you may not be needed by everyone, but it is definitely necessary for someone - your smile, your love, your strength.



:clown_face:

File: src/rotation_kind.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+
+class RotationKind(Enum):
+    ROTATION_0 = 1
+    ROTATION_90 = 2
+    ROTATION_180 = 3
+    ROTATION_270 = 4
+
+    def get_next(self) -> 'RotationKind':
+        value = self.value + 1
```
Comment (s373r) [inline src/rotation_kind.py:11]: add a blank line after

File: src/rotation_kind.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        if value > 4:
+            return RotationKind.ROTATION_0
+
+        return RotationKind(value)
+
+    def get_prev(self) -> 'RotationKind':
+        value = self.value - 1
```
Comment (s373r) [inline src/rotation_kind.py:18]: add a blank line after

File: src/rotation_kind.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+class RotationKind(Enum):
+    ROTATION_0 = 1
+    ROTATION_90 = 2
+    ROTATION_180 = 3
+    ROTATION_270 = 4
+
+    def get_next(self) -> 'RotationKind':
+        value = self.value + 1
+        if value > 4:
```
Comment (s373r) [inline src/rotation_kind.py]: use `ROTATION_270` (mb w/ `.value()`) not magic number

File: src/rotation_kind.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+            return RotationKind.ROTATION_0
+
+        return RotationKind(value)
+
+    def get_prev(self) -> 'RotationKind':
+        value = self.value - 1
+        if value == 0:
```
Comment (s373r) [inline src/rotation_kind.py]: use `ROTATION_0` (mb w/ `.value()`) not magic number

**Review by s373r — CHANGES_REQUESTED** [review]

File: src/figure_mover.py (hunk 1 above)
Comment (alena-bartosh) [reply src/figure_mover.py]: Was done before you write it 😉

File: src/figure_mover.py (hunk 2 above)
Comment (alena-bartosh) [reply src/figure_mover.py]: Done

File: src/rotation_kind.py (hunk 3 above)
Comment (alena-bartosh) [reply src/rotation_kind.py:11]: Do not agree 😉

File: src/rotation_kind.py (hunk 4 above)
Comment (alena-bartosh) [reply src/rotation_kind.py:18]: Do not agree 😉

File: src/rotation_kind.py (hunk 5 above)
Comment (alena-bartosh) [reply src/rotation_kind.py]: Tnanks!

File: src/rotation_kind.py (hunk 6 above)
Comment (alena-bartosh) [reply src/rotation_kind.py]: Done

**Review by Serg-Bartosh — APPROVED** [review]
The owner was sorry to part with a good worker, and he asked him for a favor - to build another house. The carpenter agreed, but it was clear that during the work he was thinking about something else, he didn’t try at all, he took whatever materials he got. When the carpenter finished work and the house was ready, the owner handed the carpenter the keys to the front door. “This house,” he said, “is my gift to you.” How upset the carpenter was. If he knew that he was building for himself, he would have done everything very differently.


:clown_face:

**Review by s373r — APPROVED** [review]

**Pull request merged**